src/odemis/gui/comp/overlay/ellipse.py

```diff
@@ -48,27 +47,64 @@ def __init__(self, cnvs, colour=gui.SELECTION_COLOUR):
         :param colour: (str) hex colour code for the ellipse
         """
         super().__init__(cnvs, colour)
-        # The points on the circumference of the ellipse from where the arcs are drawn
-        # The points VA is set to _points if the shape is selected
-        self._points = []
+
+    def copy(self):
+        """
+        :returns: (EllipseOverlay) a new instance of EllipseOverlay with necessary copied attributes.
+
+        """
+        shape = EllipseOverlay(self.cnvs)
+        shape.colour = self.colour
+        shape.p_point1 = self.p_point1
+        shape.p_point2 = self.p_point2
+        shape.p_point3 = self.p_point3
+        shape.p_point4 = self.p_point4
+        shape._points = self._points.copy()
```

Comment:
This whole `.points` vs `._points` is confusing. Isn't it possible to only use `.points`?  
Or at least, from the code  is seems that ._points is recomputed automatically by `.draw` from the p_points. So they are not so useful. Why should they be copied? Shouldn't `.points` (VA content) be copied instead? 
Or have an explicit function that computes the points based on the p_points? (Then `draw()` would use it to update the points, before converting them to buffer coordinates)

Reply:
I agree, it can be confusing. I am using `._points` for any internal point manipulation. `points` VA value is set post any points manipulation and if the shape is selected. Another reason is to avoid some unnecessary calculation if VA value changes often. For rectangle I can definitely get rid of `._points` and also for ellipse, but then I need it for the polygon.

In `EditableShape` its written as a comment.

```
       # list of nested points (x, y) representing the shape and whose value will be used
        # during ROA acquisition
        # The points VA is set to _points if the shape is selected
        self.points = model.ListVA()
        # Useful for internal points manipulation
        self._points: List[Vec] = []
```